server/src/Dao/AbstractDao.php

```diff
@@ -74,4 +77,35 @@ protected function paginate(
 
 		return [ $results, $continue ];
 	}
+
+	/**
+	 * Fetch all with the option of caching results
+	 *
+	 * @param QueryBuilder $qb
+	 * @param int $fetchMode
+	 * @param bool $cache
+	 * @param int $ttl
+	 * @return array
+	 */
+	protected function fetchAll(
+		QueryBuilder $qb, $fetchMode, $cache = false, $ttl = self::CONST_CACHE_TTL
+	) {
+		if ( $cache ) {
+			$qpf = new QueryCacheProfile( $ttl );
```

Comment:
Perhaps this should be in the container and injected into the class? or does this new to be created for each query?

Reply:
It needs to be created for each query. This is what creates the cache key, sets ttl, etc